build.zig

```diff
@@ -27,16 +27,28 @@ pub fn build(b: *Builder) !void {
     });
     b.default_step.dependOn(&fmt_step.step);
 
+    const test_enum = enum {
+        UNIT,
+        ALL_RUNTIME,
+        NORMAL,
+        PANIC,
+    };
+
+    comptime var available_tests: []const u8 = "";
+    inline for (@typeInfo(test_enum).Enum.fields) |field| {
+        available_tests = available_tests ++ field.name ++ ", ";
+    }
+
     const main_src = "src/kernel/kmain.zig";
     const constants_path = try fs.path.join(b.allocator, &[_][]const u8{ "src/kernel/arch", target_str, "constants.zig" });
 
     const build_mode = b.standardReleaseOptions();
-    const rt_test = b.option(bool, "rt-test", "enable/disable runtime testing") orelse false;
+    const test_type = b.option(test_enum, "test-type", "Run a specific runtime test. Available options are: " ++ available_tests) orelse test_enum.UNIT;
```

Comment:
Doesn't the test mode also determine whether or not to do unit testing not just a specific type of runtime test?

Reply:
yes, the `test_type` soon to be `test_mode` will also determin if it is to do the unit test, as the default test is `.UNIT`

Reply:
Cool, I suggest you change the help message then :)